### PR TITLE
feat: show worker name, as well as platform, teams, tags, on GUI for put/get/task steps.

### LINKED ISSUE #4517: View Worker Job Ran On
## What challenge are you facing?

Issues troubleshooting intermittent problems. We have a number of job intermittently failing and we would like to determine if this is when the jobs run on a specific worker / set of workers.

The ability to see where the job run will help us narrow down the root cause of our issues. Be that but ruling some out or by confirming some suspicions   

## What would make this better?

Within the job run output , display the worker name, ip and any other relevant data.

## Are you interested in implementing this yourself?

I would if I could, but I am not a programmer / developer.

### PR DESCRIPTION
**Pull request by evanchaoli**
Resolve #4517

Signed-off-by: Chao Li <chaol@vmware.com>

Screen shot of my test result:

show-workers

### CODE REVIEW

File: atc/exec/get_step.go [hunk 1]
```diff
@@ -189,6 +189,7 @@ func (step *GetStep) Run(ctx context.Context, state RunState) error {
 	if err != nil {
 		return err
 	}
+	step.delegate.Stderr().Write([]byte(fmt.Sprintf("Chosen worker %s: %s\n", chosenWorker.Name(), chosenWorker.Description())))
```
Comment (xtremerui) [inline atc/exec/get_step.go]: what is the reasoning behind using `stderr` vs `stdout` here? Thx.

Comment (evanchaoli) [reply atc/exec/get_step.go]: There are some discussions in PR #4398 between @vito and me. I believe Concourse generated logs (sys log) should go to stderr. Going forward, @vito is thinking to add a separate log event channel to print Concourse generated logs.

**Comment by xtremerui** [comment]
While I do like showing worker info for steps, I would like to put it into a more structure way like creating an event type for it or put the worker info into `StartGet` \ `StartPut` \ `StartTask`, so we could handle the output nicely in Web UI / Fly based on the event type. Just my 2 cents since it is more about runtime really @ddadlani .

**Comment by kcmannem** [comment]
@xtremerui I agree, UX was planning to take on https://github.com/concourse/concourse/issues/4337 and we decided to keep stderr/out clean and structure the data in the db. I don't think we should be writing this kind of info there especially since we're currently working to integrate multiple runtime backends (Kubernetes). We need a generic way to show runtime metadata.

**Comment by evanchaoli** [comment]
@xtremerui Current `StartGet` \ `StartPut` \ `StartTask` don't have a field to hold more info like this. Even if we add a field to them, it would also require UI side change to show such info. In future, some more fields, like `chosenWokrer`, could be added, but here I think what I'm adding is just a log line that shows on which workers a step will run.

**Comment by evanchaoli** [comment]
@kcmannem I guess it would take time to get features you mentioned done. Once those are out, worker info is shown with some way else, we can revert this change. But this PR will provide a quick way for OPs people to see worker info, which would be super helpful.

**Comment by kcmannem** [comment]
@evanchaoli I believe the UX team might be working on this next. Its high on their list of todos. This information is available to query from the db if needed. Users probably won't care about seeing every tasks runtime info but only care when concourse breaks. I don't want to optimize for second scenario when there is a workaround available.

**Comment by evanchaoli** [comment]
@kcmannem  Looks like #4337 is going to show initialization info with a tooltip, so I want to confirm that, as tooltip will be refreshed as initialization going forward, will all information shown by tooltip will also print to build logs?

Also, @vito, what do you think about printing chosen worker to build log?

**Comment by kcmannem** [comment]
@evanchaoli the point of #4337 is to better structure this kind of info in the db. It makes it available for both fly and the webpage to take advantage and display this UI in a more appropriate manner. I understand this is something a lot of users want (including ourselves), I've only pushed back since its something we're tackling very soon.

**Comment by vito** [comment]
@evanchaoli I don't really like the idea of printing logs into the task's output; this should be done with progress or status indicators. I would rather everyone focus on #4337 (whether we get to it or someone in the community kicks it off), as we've already got the ball rolling in terms of design and strategy. We could maybe even re-use this pattern in order to split out the image fetching logs as a separate sub-task with its own progress. At that point we'll be in a world where `stdout`/`stderr` are purely what the task printed and we've got much better progress indicators for users.

In general we try to avoid "quick fix" bugfixes or "quick and dirty" features if there's already a plan for a better design in sight. I would rather work towards the end goal, because if we adopt quick-and-dirty solutions it can easily result in de-prioritization of the better version because the need is diminished.

**Comment by evanchaoli** [comment]
OK, then I give up this PR.

**Pull request closed without merging**